File: backend/src/main/java/in/supplybase/backend/booking/BookingStatus.java
package in.supplybase.backend.booking;

/**
 * The booking lifecycle, in the order it actually happens.
 *
 * PAYMENT_PENDING is the entry point, not CONFIRMED: a booking exists the
 * moment the form is submitted, but it is not a promise to visit anyone until
 * the ₹25 has been verified server-side (RULE 7).
 */
public enum BookingStatus {
    PAYMENT_PENDING,
    BOOKING_REQUESTED,
    CONFIRMED,
    ASSIGNMENT_PENDING,
    PROFESSIONAL_ASSIGNED,
    SITE_VISIT_SCHEDULED,
    SITE_VISIT_COMPLETED,
    QUOTATION_CREATED,
    QUOTATION_SENT,
    CUSTOMER_APPROVED,
    WORK_SCHEDULED,
    WORK_IN_PROGRESS,
    WORK_COMPLETED,
    CANCELLED;

    /** Everything that has to happen before the customer sees a quotation. */
    public boolean isBeforeVisit() {
        return this == PAYMENT_PENDING || this == BOOKING_REQUESTED || this == CONFIRMED
                || this == ASSIGNMENT_PENDING || this == PROFESSIONAL_ASSIGNED
                || this == SITE_VISIT_SCHEDULED;
    }

    /** A cancelled booking is finished; nothing may move it on. */
    public boolean isFinal() {
        return this == CANCELLED || this == WORK_COMPLETED;
    }
}

File: frontend/src/lib/bookingStatus.js
/**
 * Customer-facing wording for BookingStatus (backend/.../booking/BookingStatus.java).
 * Kept separate from the admin app's own internal labels — a client sees
 * "Assigning a Professional", staff see "ASSIGNMENT_PENDING".
 */
const LABELS = {
  PAYMENT_PENDING: 'Payment Pending',
  BOOKING_REQUESTED: 'Booking Requested',
  CONFIRMED: 'Confirmed',
  ASSIGNMENT_PENDING: 'Assigning a Professional',
  PROFESSIONAL_ASSIGNED: 'Professional Assigned',
  SITE_VISIT_SCHEDULED: 'Site Visit Scheduled',
  SITE_VISIT_COMPLETED: 'Site Visit Completed',
  QUOTATION_CREATED: 'Quotation Ready',
  QUOTATION_SENT: 'Quotation Sent',
  CUSTOMER_APPROVED: 'Quotation Approved',
  WORK_SCHEDULED: 'Work Scheduled',
  WORK_IN_PROGRESS: 'Work In Progress',
  WORK_COMPLETED: 'Completed',
  CANCELLED: 'Cancelled',
};

export const bookingStatusLabel = (status) => LABELS[status] || status;

/** Same three-tone split as the admin app's toneFor(), for one visual language. */
export const bookingStatusTone = (status) => {
  if (status === 'CANCELLED') return 'danger';
  if (status === 'WORK_COMPLETED') return 'success';
  if (['PAYMENT_PENDING', 'BOOKING_REQUESTED', 'ASSIGNMENT_PENDING'].includes(status)) return 'warning';
  return 'accent';
};
